edward/util/random_variables.py

```diff
@@ -699,3 +699,52 @@ def get_variables(x, collection=None):
     nodes.update(node.op.inputs)
 
   return list(output)
+
+
+def softmax(vec):
+  numerator = np.exp(vec)
+  return numerator / numerator.sum(axis=0)
+
+
+def compute_multinomial_mode(probs, total_count=1):
```

Comment:
Is there motivation for writing this function in NumPy? In `evaluate.py`, it seems like you could keep `probs` as a `tf.Tensor` and make this function return the multinomial mode as a `tf.Tensor`.

Reply:
Oh man. I tried for 2 days to implement this in TensorFlow.

Both the `while` loop — and updating `tf.Variable`s within — and the `np.random.choice` bit were *very* icky/overly-complicated in TensorFlow. 

I then realized that the only parameter this accepts is a vector. I asked myself if this vector (a probability distribution) would ever be exceptionally large, to which I answered unlikely. Finally, and at least for a first pass, I resolved that this vector would generally not be a multi-dimensional Tensor.

So, I went with Numpy, which made everything exceedingly straightforward.

Reply:
Ah, yes that makes a lot of sense. NumPy is okay.